File: 2145. Count the Hidden Sequences/numberOfArrays.cpp
using namespace std;
#include <iostream>
#include <vector>

class Solution {
public:
    int numberOfArrays(vector<int>& differences, int lower, int upper) {
        int max = lower;
        int left = 0 ; //Pointer
        
        int ans = lower;
        int start = lower;

        //get the started array
        while(left < differences.size()){
            int sum = ans + differences[left];
            if(sum > upper)return 0;
            if(sum >= lower && sum <= upper){
                ans = sum;
                left++;
                if(sum > max){
                    max = sum;
                }
            }
            else{
                left = 0;
                start++;
                ans = start;
                max = start;
            }
        }
        return (upper - max) + 1;
    }
};

File: 2145. Count the Hidden Sequences/numberOfArrays.js
/**
 * @param {number[]} differences
 * @param {number} lower
 * @param {number} upper
 * @return {number}
 */
var numberOfArrays = function (differences, lower, upper) {
    let left = 0
    let max = lower;

    let ans = lower
    let start = lower

    while (left < differences.length) {               //O(n)
        let sum = ans + differences[left]
        if (sum > upper) return 0;
        if (sum >= lower && sum <= upper) {
            ans = sum
            left++
            max = Math.max(max, ans)
        }
        else {
            left = 0
            start++;
            ans = start;
            max = start
        }
    }
    return (upper - max) + 1
};